core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1446,11 +1448,20 @@ class KafkaApis(val requestChannel: RequestChannel,
       groupCoordinator.handleHeartbeat(
         heartbeatRequest.data.groupId,
         heartbeatRequest.data.memberId,
+        getGroupInstanceId(heartbeatRequest.data.groupInstanceId),
         heartbeatRequest.data.generationid,
         sendResponseCallback)
     }
   }
 
+  def getGroupInstanceId(rawInstanceId: String): Option[String] = {
+      if (rawInstanceId == null ||
+        config.interBrokerProtocolVersion < KAFKA_2_3_IV0)
+        None
+      else
+        Some(rawInstanceId)
```

Comment:
Should this be `Option(rawInstanceId)`. The group instance id could be null, right?

Reply:
Checked null in the above if statement.

Reply:
Ack. Read through a bit too quickly.